src/client/datascience/editor-integration/codeLensFactory.ts

```diff
@@ -354,14 +354,7 @@ export class CodeLensFactory implements ICodeLensFactory, IInteractiveWindowList
         isFirst: boolean
     ): CodeLens | undefined {
         // Do not generate interactive window codelenses for TextDocuments which are part of NotebookDocuments
-        // and do not generate debug codelenses for new interactive window (not properly supported yet)
-        if (
-            (commandName.includes('debug') &&
-                workspace.getConfiguration('interactive.experiments')?.get('enable') === true) ||
-            workspace.notebookDocuments.find(
-                (notebook) => notebook.uri.toString() === document.notebook?.uri.toString()
-            )
-        ) {
+        if (workspace.notebookDocuments.find((notebook) => notebook.uri.toString() === document.uri.toString())) {
```

Comment:
Isn't this the same? I feel its better.
```suggestion
        if (workspace.notebookDocuments.find((notebook) => notebook === document)) {
```

Reply:
IIRC they are not guaranteed to be the same object, so `document.uri.toString()` is the recommended way to check for document equality.

Reply:
There should be no duplicated documents (`TextDocument`, `NotebookDocument`) in the exthost.

Reply:
There is a scheme vscode-notebook-cell for the URI on NotebookCell textDocuments. That might be an easier check.

Reply:
> IIRC they are not guaranteed to be the same object, so document.uri.toString() is the recommended way to check for document equality.

Whoa, @rebornix is that correct. We use references so over today in Jupyter, weakmaps in a number of places. Ie we awesome there's only one insurance of the same document, hence same ref.

Reply:
Just to close the loop here, you can't directly compare URI objects for equality since they're not guaranteed to be the same object, but comparing TextDocuments and NotebookDocuments should be safe

Reply:
Nope they are guaranteed to the be same object, thats how it works today.
We use instances in Native Notebooks, instead of the Uri. VS Code never re-create the notebook object.